colcon_core/which_python.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Dan Rose, no rights reserved
+# Licensed under the Apache License, Version 2.0
+
+from colcon_core.logging import colcon_logger
+from colcon_core.environment_variable import EnvironmentVariable
+import os
+import shutil
+
+logger = colcon_logger.getChild(__name__)
+
+"""Environment variable to set the python executable"""
+PYTHON_EXECUTABLE_ENVIRONMENT_VARIABLE = EnvironmentVariable(
+    'COLCON_PYTHON_EXECUTABLE',
+    'Set the python executable (default: `which python3`)')
+
+
+def which_python():
```

Comment:
Please add a unit test which covers 100% of this new function. (This can wait until all other feedback has been addressed and CI is passing to avoid multiple iterations.)